README.md

```diff
@@ -1,68 +1,73 @@
+are more likely to at least have some issues when used with PGAdapter.
 
 ## Usage
-The PostgreSQL adapter can be started both as a Docker container, a standalone process as well as an in-process server.
-
+The PostgreSQL adapter can be started both as a Docker container, a standalone process as well as an
+in-process server (the latter is only supported for Java applications).
 
 ### Docker
 
+See [running PGAdapter using Docker](docs/docker.md) for more examples for running PGAdapter in Docker.
+
 Replace the project, instance and database names and the credentials file in the example below to
 run PGAdapter from a pre-built Docker image.
 
 ```shell
+export GOOGLE_APPLICATION_CREDENTIALS=/path/to/credentials.json
 docker pull us-west1-docker.pkg.dev/cloud-spanner-pg-adapter/pgadapter-docker-images/pgadapter
 docker run \
   -d -p 5432:5432 \
-  -v /local/path/to/credentials.json:/tmp/keys/key.json:ro \
-  -e GOOGLE_APPLICATION_CREDENTIALS=/tmp/keys/key.json \
-  us-west1-docker.pkg.dev/cloud-spanner-pg-adapter/pgadapter/pgadapter \
+  -v ${GOOGLE_APPLICATION_CREDENTIALS}:${GOOGLE_APPLICATION_CREDENTIALS}:ro \
+  -e GOOGLE_APPLICATION_CREDENTIALS \
+  us-west1-docker.pkg.dev/cloud-spanner-pg-adapter/pgadapter-docker-images/pgadapter \
   -p my-project -i my-instance -d my-database \
   -x
 ```
 
-The option `-v /local/path/to/credentials.json:/tmp/keys/key.json:ro` mounts the credentials file on
-your local system to a file in the Docker container. The `-e GOOGLE_APPLICATION_CREDENTIALS=/tmp/keys/key.json`
-sets the file as the default credentials to use inside the container.
+The option `-v ${GOOGLE_APPLICATION_CREDENTIALS}:${GOOGLE_APPLICATION_CREDENTIALS}:ro` mounts the
+default credentials file on your local system to a virtual file with the same name in the Docker container.
+The `-e GOOGLE_APPLICATION_CREDENTIALS` sets the file as the default credentials to use inside the container.
+
+The `-x` option tells PGAdapter to accept connections other hosts than `localhost`. This is required
+when running PGAdapter in a Docker container, as requests will not be seen as coming from `localhost`
+in the container. Choose a different host port than 5432 if you already have PostgreSQL running on
+your local system.
+
+See [Options](#Options) for an explanation of all further options.
+
+### Standalone with pre-built jar
+
+A pre-built jar containing all dependencies can be downloaded from https://storage.googleapis.com/pgadapter-jar-releases/pgadapter-VERSION.jar
+
+Example (replace `v0.2.1` with the version you want to download):
+
+```shell
+VERSION=v0.2.1
+wget https://storage.googleapis.com/pgadapter-jar-releases/pgadapter-${VERSION}.jar
+java -jar pgadapter-${VERSION}.jar -p my-project -i my-instance -d my-database
+```
+
+Use the `-s` option to specify a different local port than the default 5432 if you already have
```

Comment:
I think we should keep it here, as it is a very typical problem that (first time) users will run into when running the uber-jar. Many people who will try this out will already have a local PG installation, and I think that this section should be as complete as possible for anyone who wants to try it out, without having to go through the complete list of options to figure out how to use a custom port. Especially as PGAdapter uses `-s` instead of `-p` to specify the port to use.

Reply:
ack